src/daft-local-execution/Cargo.toml

```diff
@@ -40,6 +40,9 @@ tokio = {workspace = true}
 tokio-util = {workspace = true}
 tracing = {workspace = true}
 
+[dev-dependencies]
+daft-functions = {path = "../daft-functions", default-features = false}
```

Comment:
non-blocking, but I believe @universalmind303 would prefer taking a dependency here only on the necessary package (correct me if I'm wrong).

For now, I think you must depend on all of daft-functions until the url_download / get_resource_request is split out. Since this fixes CI builds, not blocking - long term it would be nice to reduce the scope of the import, but again I may not have the full picture here.

https://github.com/Eventual-Inc/Daft/commit/567757652dc09ff98b87e0403ce73d1bedd909fc#diff-6acc4f7e1868a8ec0a639dfc3712d31e5bd83ae0d7f7cf1fe93ac90225782d5e

Reply:
That's fair. At some point, we may even turn the url_download function into a completely separate physical operator for performance reasons, but this was a temporary fix.

Reply:
For sure, the lower we push it the better.